apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -299,6 +280,50 @@ defmodule Aecore.Miner.Worker do
     end)
   end
 
+  defp filter_transactions_by_block_size(txs, new_block_size_bytes) do
+    {valid_txs_by_block_size, _} =
+      List.foldl(txs, {[], false}, fn(tx, {selected_txs, limit_reached}) ->
+        if !limit_reached do
+          new_selected_txs = [tx | selected_txs]
+          new_selected_txs_size_bits =
+            new_selected_txs |> :erlang.term_to_binary() |> Bits.extract() |> Enum.count()
```

Comment:
You should use `byte_size/1` function instead of `Bits.extract() |> Enum.count()` it should be much faster.